src/test/java/winstone/HttpConnectorFactoryTest.java

```diff
@@ -49,6 +56,29 @@ public void testListenAddress() throws Exception {
 
     @Test
     public void writePortInFile() throws Exception {
+        Callable<String> asyncVerifier = () -> {
+            System.out.println("Async Verifier started!");
+
+            while (true) {
+
+                Path portFile = Paths.get(tmp.getRoot().getAbsolutePath(), "subdir/jenkins.port");
+                if (Files.exists(portFile)) {
+                    try {
+                        return FileUtils.readFileToString(portFile.toFile(), StandardCharsets.UTF_8);
+                    } catch (IOException e) {
+                        throw new RuntimeException(e);
+                    }
+                }
+                try {
+                    Thread.sleep(1);
+                } catch (InterruptedException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        };
+        ExecutorService executorService = Executors.newSingleThreadExecutor();
+        Future<String> futurePort = executorService.submit(asyncVerifier);
```

Comment:
using something like this would result in cleaner code:
http://www.awaitility.org/

Reply:
Thanks, applied in [8c54050](https://github.com/jenkinsci/winstone/pull/249/commits/8c540507e1e5797aac7189e09d1542bb672532a3)